src/libraries/Microsoft.Extensions.Http/tests/Microsoft.Extensions.Http.Tests/Logging/LoggingUriOutputTests.cs

```diff
@@ -16,156 +16,198 @@ namespace Microsoft.Extensions.Http.Tests.Logging
-            // Arrange
-            var sink = new TestSink();
+            public static readonly EventId RequestStart = new EventId(100, "RequestStart");
+            public static readonly EventId RequestEnd = new EventId(101, "RequestEnd");
 
-            var serviceCollection = new ServiceCollection();
-            serviceCollection.AddLogging();
-            serviceCollection.AddSingleton<ILoggerFactory>(new TestLoggerFactory(sink, enabled: true));
-
-            serviceCollection
-            .AddHttpClient("test")
-            .ConfigurePrimaryHttpMessageHandler(() => new TestMessageHandler());
-
-            var services = serviceCollection.BuildServiceProvider();
-
-            var client = services.GetRequiredService<IHttpClientFactory>().CreateClient("test");
-
-
-            // Act
-            var request = new HttpRequestMessage(HttpMethod.Get, "http://api.example.com/search?term=Western%20Australia");
-
-            await client.SendAsync(request);
-
-            // Assert
-            var messages = sink.Writes.ToArray();
-
-            var message = Assert.Single(messages.Where(m =>
-            {
-                return
-                    m.EventId == LoggingHttpMessageHandler.Log.EventIds.RequestStart &&
-                    m.LoggerName == "System.Net.Http.HttpClient.test.ClientHandler";
-            }));
-
-            Assert.Equal("Sending HTTP request GET http://api.example.com/search?term=Western%20Australia", message.Message);
+            public static readonly EventId PipelineStart = new EventId(100, "RequestPipelineStart");
+            public static readonly EventId PipelineEnd = new EventId(101, "RequestPipelineEnd");
         }
 
-        [Fact]
-        public async Task LoggingScopeHttpMessageHandler_LogsAbsoluteUri()
+        public static readonly TheoryData<string, string> GetRedactedUriString_Data = new TheoryData<string, string>()
         {
-            // Arrange
-            var sink = new TestSink();
-
-            var serviceCollection = new ServiceCollection();
-            serviceCollection.AddLogging();
-            serviceCollection.AddSingleton<ILoggerFactory>(new TestLoggerFactory(sink, enabled: true));
-
-            serviceCollection
-            .AddHttpClient("test")
-            .ConfigurePrimaryHttpMessageHandler(() => new TestMessageHandler());
-
-            var services = serviceCollection.BuildServiceProvider();
-
-            var client = services.GetRequiredService<IHttpClientFactory>().CreateClient("test");
-
-
-            // Act
-            var request = new HttpRequestMessage(HttpMethod.Get, "http://api.example.com/search?term=Western%20Australia");
-
-            await client.SendAsync(request);
+            { null, null },
+            { "http://q.app/foo", "http://q.app/foo" },
+            { "http://q.app/foo?", "http://q.app/foo?" },
+            { "http://q.app/foo?XXX", "http://q.app/foo?*" },
+            { "http://q.app/a/b/c?a=b%20c&x=1", "http://q.app/a/b/c?*" },
```

Comment:
Unlike with headers, where AFAIK we only replace the hader values, we replace the entire query string with a `*`.

We could log `?a=*&x=*` here, but it would come with a parsing price I assume we don't want to pay for the default log level.